### PR TITLE
[IOCIT-199]  appbackend too many errors alert

### PR DESCRIPTION
**Pull request by gquadrati**
### List of changes

Add "Too many errors" alert for io-p-app-backendl(1-2-i)

### Motivation and context

Alerts refactoring

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/core/99_variables.tf [hunk 1]
```diff
@@ -443,6 +455,13 @@ variable "eucovidcert_alerts_enabled" {
 }
 
 # app backend
+
+variable "app_backend_names" {
+  description = "App backend instance names"
+  type        = list(string)
+  default     = ["appbackendl1", "appbackendl2", "appbackendli"]
```
Comment (diegolagospagopa) [inline src/core/99_variables.tf:462]: please can you put this values into the env folder

File: src/core/app_backend.tf [hunk 2]
```diff
@@ -1012,3 +1012,44 @@ module "app_backend_web_test_api" {
...
 
 }
+
+# -----------------------------------------------
+# Alerts
+# -----------------------------------------------
+
+data "azurerm_app_service" "app_backend_app_services" {
+  for_each            = toset(var.app_backend_names)
+  name                = format("%s-app-${each.value}", local.project)
```
Comment (diegolagospagopa) [inline src/core/app_backend.tf:1022]: please use only the interpolation format

File: src/core/app_backend.tf [hunk 3]
```diff
@@ -1012,3 +1012,44 @@ module "app_backend_web_test_api" {
...
+  resource_group_name = azurerm_resource_group.rg_linux.name
+}
+
+resource "azurerm_monitor_metric_alert" "too_many_error" {
```
Comment (diegolagospagopa) [inline src/core/app_backend.tf:1026]: can you find a better name for: "too_many_error"

File: src/core/app_backend.tf [hunk 4]
```diff
@@ -1012,3 +1012,44 @@ module "app_backend_web_test_api" {
...
+  for_each = { for key, name in data.azurerm_app_service.app_backend_app_services : key => name }
+
+  name                = "[IO-COMMONS | ${each.value.name}] Too many errors"
+  resource_group_name = azurerm_resource_group.rg_linux.name
+  scopes              = [each.value.id]
+  # TODO: add Runbook for checking errors
+  description   = "Whenever the total http server errors exceeds a dynamic threashold. Runbook: TODO"
+  severity      = 0
+  window_size   = "PT5M"
+  frequency     = "PT5M"
```
Comment (diegolagospagopa) [inline src/core/app_backend.tf:1036]: may be 5 minutes is too short and false positives can arise?

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]

File: src/core/app_backend.tf (hunk 3 above)
Comment (gquadrati) [reply src/core/app_backend.tf]: `too_may_http_5xx` is better?

Comment (gquadrati) [reply src/core/app_backend.tf]: Done in [fix after review](https://github.com/pagopa/io-infra/pull/296/commits/d04ab9cea130bcf3dcbebbf4d7a819d4ab95b1d1)

File: src/core/app_backend.tf (hunk 2 above)
Comment (gquadrati) [reply src/core/app_backend.tf]: Right. Done in [fix after review](https://github.com/pagopa/io-infra/pull/296/commits/d04ab9cea130bcf3dcbebbf4d7a819d4ab95b1d1)

File: src/core/99_variables.tf (hunk 1 above)
Comment (gquadrati) [reply src/core/99_variables.tf]: Sure, done in [fix after review](https://github.com/pagopa/io-infra/pull/296/commits/d04ab9cea130bcf3dcbebbf4d7a819d4ab95b1d1)

File: src/core/app_backend.tf (hunk 4 above)
Comment (diegolagospagopa) [reply src/core/app_backend.tf:1036]: ok great

**Review by diegolagospagopa — DISMISSED** [review]

**Review by fabriziopapi — DISMISSED** [review]

**Review by fabriziopapi — DISMISSED** [review]

**Review by fabriziopapi — APPROVED** [review]

**Pull request merged**